python/GafferSceneUI/SetFilterUI.py

```diff
@@ -94,3 +98,148 @@
 	}
 
 )
+
+##########################################################################
+# NodeGadget drop handler
+##########################################################################
+
+__DropMode = enum.Enum( "__DropMode", [ "None_", "Add", "Remove", "Replace" ] )
+
+__originalDragPointer = None
+
+def __setsPlug( node ) :
+
+	for plug in node.children( Gaffer.Plug ) :
```

Comment:
This wasn't part of my original rationale, but in light of https://github.com/GafferHQ/gaffer/pull/5796, do we want to recurse? If we end up merging that and applying the same logic here, recursing could find an unpromoted plug inside a Box / Reference?

Reply:
The RecursiveRanges use a predicate to determine when recursion occurs, and the Plug variants ensure recursion doesn't disappear into nodes-inside-nodes. In practice whether we recurse or not might not be important, but I think it's worth using at least `Plug.Range` as our canonical way of iterating. Should probably be `InputRange` as well, since we can't set output plugs?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5786/commits/095f25773d5e5ea7f159487ca55562dba0ad74e3